packages/app/src/EthereumProviders.tsx

```diff
@@ -5,10 +5,27 @@ import { chain, configureChains, createClient, WagmiConfig } from "wagmi";
 import { alchemyProvider } from "wagmi/providers/alchemy";
 import { publicProvider } from "wagmi/providers/public";
 
-export const targetChainId = parseInt(process.env.CHAIN_ID || "0") || 5;
+export const targetChainId =
+  parseInt(process.env.NEXT_PUBLIC_CHAIN_ID || "0") || 5;
+const targetChains = [chain.mainnet];
+switch (targetChainId) {
+  case 3:
+    targetChains.push(chain.ropsten);
+    break;
+  case 4:
+    targetChains.push(chain.rinkeby);
+    break;
+  case 42:
+    targetChains.push(chain.kovan);
+    break;
+  case 5:
+  default:
+    targetChains.push(chain.goerli);
+    break;
+}
```

Comment:
Since the `chain` object contains the `id`, maybe worth doing something like

```jsx
import { defaultChains } from 'wagmi';

export const targetChainId = parseInt(process.env.NEXT_PUBLIC_CHAIN_ID || "0") || 5;

// filter down to just mainnet + optional target testnet chain so that rainbowkit can tell
// the user to switch network if they're on an alternative one
const targetChains = defaultChains.filter((chain) => chain.id === 1 || chain.id === targetChainId)
```

Reply:
This is way cleaner than what I wrote after being up all night with the baby : )